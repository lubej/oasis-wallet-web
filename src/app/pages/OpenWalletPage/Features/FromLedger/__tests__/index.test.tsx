import * as React from 'react'
import { render, act, screen } from '@testing-library/react'
import { FromLedger } from '..'
import { MemoryRouter } from 'react-router-dom'

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}))

jest.mock('../../../../../lib/ledger', () => ({
  ...jest.requireActual('../../../../../lib/ledger'),
  // Throws BLE not supported
  canAccessBle: () => jest.fn().mockReturnValue(false),
}))

const renderComponent = () =>
  render(
    <MemoryRouter>
      <FromLedger />
    </MemoryRouter>,
  )

describe('<FromLedger />', () => {
  it('should render component', async () => {
    const { container } = renderComponent()

    await act(() => Promise.resolve())

    expect(container).toMatchSnapshot()

    expect(screen.getByText('errors.usbTransportNotSupported')).toBeInTheDocument()
    expect(screen.getByText('errors.bluetoothTransportNotSupported')).toBeInTheDocument()
  })

  it('should render component with an ledger access button', async () => {
    renderComponent()

    await act(() => Promise.resolve())

    expect(screen.queryByText('openWallet.importAccounts.usbLedger')).toBeInTheDocument()
    expect(screen.queryByText('openWallet.importAccounts.bluetoothLedger')).toBeInTheDocument()
  })
})
