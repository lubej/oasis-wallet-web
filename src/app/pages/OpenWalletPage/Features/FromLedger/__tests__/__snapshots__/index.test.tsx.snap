// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FromLedger /> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 12px;
  background-color: #FFFFFF;
  color: #444444;
  border: solid 1px background-front-border;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c5 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c1 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c3:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus > circle,
.c3:focus > ellipse,
.c3:focus > line,
.c3:focus > path,
.c3:focus > polygon,
.c3:focus > polyline,
.c3:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c3:focus::-moz-focus-inner {
  border: 0;
}

.c3:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible) > circle,.c3:focus:not(:focus-visible) > ellipse,
.c3:focus:not(:focus-visible) > line,.c3:focus:not(:focus-visible) > path,
.c3:focus:not(:focus-visible) > polygon,.c3:focus:not(:focus-visible) > polyline,
.c3:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c3:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    border: solid 1px background-front-border;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    width: auto;
    height: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      openWallet.importAccounts.connectDeviceHeader
    </h1>
    <div
      class="c2"
    >
      <div>
        <div>
          <span>
            <a
              aria-label="openWallet.importAccounts.usbLedger"
              href="/usb"
            >
              <button
                class="c3"
                disabled=""
                tabindex="-1"
                type="button"
              >
                openWallet.importAccounts.usbLedger
              </button>
            </a>
          </span>
        </div>
        <span
          class="c4"
        >
          errors.usbTransportNotSupported
        </span>
      </div>
      <div
        class="c5"
      />
      <div>
        <div>
          <a
            aria-label="openWallet.importAccounts.bluetoothLedger"
            href="/ble"
          >
            <button
              class="c3"
              disabled=""
              tabindex="-1"
              type="button"
            >
              openWallet.importAccounts.bluetoothLedger
            </button>
          </a>
        </div>
        <span
          class="c4"
        >
          errors.bluetoothTransportNotSupported
        </span>
      </div>
    </div>
  </div>
</div>
`;
